### PR TITLE
[BROWSEUI] Clear pidlLastParsed on CAddressEditBox::Execute

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Based on KRosUser's `gopidlparsed.patch`.
JIRA issue: [CORE-19019](https://jira.reactos.org/browse/CORE-19019)

## Proposed changes

- Clear `pidlLastParsed` if necessary.

## TODO

- [x] Do tests.

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]
Approved. Please wait for a 2nd approval before merging.

File: dll/win32/browseui/addresseditbox.cpp
Comment (learn-more) [inline dll/win32/browseui/addresseditbox.cpp:322]: Comparing against `0` is cleaner here imo.
Apart from that, I don't know why there was code storing the last parsed pidl, and how that is influenced by this change.
@JoachimHenze can you elaborate on that?

Comment (katahiromz) [reply dll/win32/browseui/addresseditbox.cpp:322]:
```suggestion
    if (hr == 0)
```

Comment (JoachimHenze) [reply dll/win32/browseui/addresseditbox.cpp]: No I can't elaborate on that. The code is from @KrosUser. Not from me. I gave my approval based on the promised functionality, and because I could not spot formal issues in the suggested patch.

Comment (HBelusca) [reply dll/win32/browseui/addresseditbox.cpp]: > Comparing against 0 is cleaner here imo.

What's the reasoning behind this?

**Comment by HBelusca** [comment]
Can someone explain to me why the `pidlLastParsed` needs to be cleared only in this exit path and not the other ones in this function? (for example the `SUCCEEDED(hr)` case of the call `pisb->BrowseObject(pidlLastParsed, 0);` that is just below, or others...)

File: dll/win32/browseui/addresseditbox.cpp
Comment (learn-more) [reply dll/win32/browseui/addresseditbox.cpp]: Because it's a comparison result.

Comment (HBelusca) [reply dll/win32/browseui/addresseditbox.cpp]: ah ok, makes sense then

**Comment by katahiromz** [comment]
> Can someone explain to me why the pidlLastParsed needs to be cleared only in this exit path

When `pidlLastParsed != NULL`, it means parsing is done.
`CAddressEditBox::Execute` should clear `pidlLastParsed` if parsing is done.
If not doing so, `CAddressEditBox::Execute` couldn't detect the change of address bar edit box.

**Pull request merged**
